src/System.Management/src/System/Management/ManagementException.cs

```diff
@@ -0,0 +1,799 @@
+			}
+			catch {}
+		}
+
+        /// <summary>
+        /// <para>Initializes a new instance of the <see cref='System.Management.ManagementException'/> class that is serializable.</para>
+        /// </summary>
+        /// <param name='info'>The <see cref='System.Runtime.Serialization.SerializationInfo'/> to populate with data.</param>
+	/// <param name='context'>The destination (see <see cref='System.Runtime.Serialization.StreamingContext'/> ) for this serialization.</param>
+        protected ManagementException(SerializationInfo info, StreamingContext context) : base(info, context)
+        {
+            errorCode = (ManagementStatus)info.GetValue("errorCode", typeof(ManagementStatus));
+            errorObject = info.GetValue("errorObject", typeof(ManagementBaseObject)) as ManagementBaseObject;
+        }
+		/// <summary>
+		/// <para>Initializes a new instance of the <see cref='System.Management.ManagementException'/> class</para>
+		/// </summary>
+		public ManagementException():this(ManagementStatus.Failed, "", null)
+		{
+			 
+		}
+
+		
+		/// <summary>
+		/// <para>Initializes a new instance of the <see cref='System.Management.ManagementException'/> 
+		/// class with a specified error message.</para>
+		/// <param name='message'>The message that describes the error.</param>
+		/// </summary>
+		public ManagementException(string message):this(ManagementStatus.Failed, message, null)
+		{
+			
+		}
+
+		/// <summary>
+		/// <para>Initializes a empty new instance of the <see cref='System.Management.ManagementException'/> class </para>
+		/// <param name='message'>The message that describes the error.</param>
+		/// <param name='innerException'>The exception that is the cause of the current exception. If the innerException 
+		/// parameter is not a null reference (Nothing in Visual Basic), the current exception is raised in a catch 
+		/// block that handles the inner exception.</param>
+		/// </summary>
+		public ManagementException(string message,Exception innerException):this(innerException, message, null)
+		{	
+			// if the exception passed is not a ManagementException, then initialize the ErrorCode to Failed
+			if (!(innerException is ManagementException))
+				errorCode = ManagementStatus.Failed;
+		}
+
+        /// <summary>
+        /// <para>Populates the <see cref='System.Runtime.Serialization.SerializationInfo'/> object with the data needed to 
+        ///    serialize the <see cref='System.Management.ManagementException'/>
+        ///    object.</para>
+        /// </summary>
+        /// <param name='info'>The <see cref='System.Runtime.Serialization.SerializationInfo'/> to populate with data.</param>
+	/// <param name='context'>The destination (see <see cref='System.Runtime.Serialization.StreamingContext'/> ) for this serialization.</param>
+ 
+		//FXCop requests explicit demand of the SerializationFormatter permission
+		[SecurityPermission(SecurityAction.Demand, SerializationFormatter=true)]
```

Comment:
you can remove all security permission stuff declarative and imperative. It does nothing on core.

Reply:
Done